File: uikit/src/main/java/com/sendbird/uikit/activities/viewholder/AdminMessageViewHolder.java
package com.sendbird.uikit.activities.viewholder;

import android.view.View;

import androidx.annotation.NonNull;

import com.sendbird.android.channel.BaseChannel;
import com.sendbird.android.message.BaseMessage;
import com.sendbird.uikit.consts.MessageGroupType;
import com.sendbird.uikit.databinding.SbViewAdminMessageBinding;
import com.sendbird.uikit.internal.ui.messages.AdminMessageView;

import java.util.Map;

public final class AdminMessageViewHolder extends MessageViewHolder {
    @NonNull
    private final AdminMessageView adminMessageView;

    AdminMessageViewHolder(@NonNull SbViewAdminMessageBinding binding, boolean useMessageGroupUI) {
        super(binding.getRoot(), useMessageGroupUI);
        adminMessageView = binding.adminMessageView;
    }

    @Override
    public void bind(@NonNull BaseChannel channel, @NonNull BaseMessage message, @NonNull MessageGroupType messageGroupType) {
        adminMessageView.setMessageUIConfig(messageUIConfig);
        adminMessageView.drawMessage(message);
    }

    @Override
    @NonNull
    public Map<String, View> getClickableViewMap() {
        return clickableViewMap;
    }
}

File: uikit/src/main/java/com/sendbird/uikit/internal/ui/viewholders/AdminMessageViewHolder.kt
package com.sendbird.uikit.internal.ui.viewholders

import android.view.View
import com.sendbird.android.channel.BaseChannel
import com.sendbird.android.message.BaseMessage
import com.sendbird.uikit.activities.viewholder.MessageViewHolder
import com.sendbird.uikit.databinding.SbViewAdminMessageBinding
import com.sendbird.uikit.internal.ui.messages.AdminMessageView
import com.sendbird.uikit.model.MessageListUIParams

internal class AdminMessageViewHolder constructor(
    binding: SbViewAdminMessageBinding,
    messageListUIParams: MessageListUIParams
) : MessageViewHolder(binding.root, messageListUIParams) {
    private val adminMessageView: AdminMessageView

    init {
        adminMessageView = binding.adminMessageView
    }

    override fun bind(channel: BaseChannel, message: BaseMessage, params: MessageListUIParams) {
        adminMessageView.messageUIConfig = messageUIConfig
        adminMessageView.drawMessage(message)
    }

    override fun getClickableViewMap(): Map<String, View> = mapOf()
}
